spec/unit/type/rds_instance_spec.rb

```diff
@@ -108,4 +109,41 @@
     end
   end
 
+  it 'backup retention must be an integer' do
+    expect {
+      type_class.new(:name => 'sample', :backup_retention_period => 'Ten')
+    }.to raise_error(Puppet::ResourceError, /backup_retention_period must be an integer/)
+  end
+
+  it 'should be happy with strings for backup_retention_period' do
+    machine = type_class.new(:name => 'sample', :backup_retention_period => 40)
+    expect(machine.property(:backup_retention_period).insync?('40')).to be true
+  end
+
+  it 'should be happy with integers for backup_retention_period' do
+    machine = type_class.new(:name => 'sample', :backup_retention_period => 45)
+    expect(machine.property(:backup_retention_period).insync?(45)).to be true
+  end
```

Comment:
This and the prior test case could be joined into a common context:

```
  context 'with the backup_retention_period set to an integer' do
    let(:machine) { type_class.new(:name => 'sample', :backup_retention_period => 40) }

    it 'should be happy with strings for backup_retention_period' do
      expect(machine.property(:backup_retention_period).insync?('40')).to be true
    end

    it 'should be happy with integers for backup_retention_period' do
      expect(machine.property(:backup_retention_period).insync?(40)).to be true
    end
  end
```

The same could be done for all the default test cases, and probably for a number of the happy-path insync tests to share a single instance.